tools/metadata_generation/swapi_metadata_generator.py

```diff
@@ -0,0 +1,97 @@
+
+import json
+
+import pandas as pd
+
+
+def get_global_attributes(sheet):
+    """Get the global attributes from a metadata spreadsheet.
+
+    Parameters
+    ----------
+    sheet : pandas.DataFrame
+        A sheet from an Excel file represented as a pandas DataFrame.
+
+    Returns
+    -------
+    dict
+        A dictionary of the global attributes, where the keys are the attribute names
+        and the values are the attribute descriptions.
+    """
+    sheet = sheet.set_index("NAME")
+    # subset by global_att field type, then only keep the CATDESC column
+    sheet = sheet[sheet["FIELD_TYPE"] == "global_att"][["CATDESC"]]
+    return sheet.to_dict()["CATDESC"]
+
+
+def get_file_attributes(sheet, global_attrs=None):
+    """Get the file attributes from a metadata spreadsheet.
+
+    Parameters
+    ----------
+    sheet : pandas.DataFrame
+        A sheet from an Excel file represented as a pandas DataFrame.
+
+    Returns
+    -------
+    dict
+        A dictionary of the file attributes, where the keys are the attribute names
+        and the values are the attribute descriptions.
+    """
+    # Add the global attributes for this specific sheet
+    file_attrs = {"global_attrs": get_global_attributes(sheet)}
+    # Add the global attributes from the global_attrs sheet (if any)
+    file_attrs["global_attrs"].update(global_attrs or {})
+
+    # Move onto variable and coordinate metadata
+    # Get everything that isn't a global attribute and drop that column
+    sheet = sheet[sheet["FIELD_TYPE"] != "global_att"].drop(columns=["FIELD_TYPE"])
+    sheet = sheet.set_index("NAME")
+
+    # Iterate through the rows of the sheet, each row being a variable or coordinate
+    # Then drop any NaN values and convert the row to a dictionary of key/values like:
+    # {"CATDESC": "description", "VAR_TYPE": "support_data", ...}
+    for row, data in sheet.iterrows():
+        file_attrs[row] = data.dropna().to_dict()
+
+    return file_attrs
+
+
+# Load all sheets
+def process_file(excel_path, output_json_path):
+    """Process the metadata file and output the metadata to a JSON file.
+
+    Parameters
+    ----------
+    excel_path : str | pathlib.Path
+        The path to the metadata Excel file.
+    output_json_path : str | pathlib.Path
+        The path to the output JSON file.
+    """
+    sheets = pd.read_excel(excel_path, sheet_name=None)
+
+    # Dictionary to store all of the metadata
+    metadata_dict = {}
+    metadata_dict["global_attrs"] = get_global_attributes(sheets["global_attrs"])
+
+    for key in sheets:
+        if not key.startswith("imap_"):
+            # Only worry about the imap_* sheets
+            # ignore support data, and codes
+            continue
+        metadata_dict[key] = get_file_attributes(
+            sheets[key], metadata_dict["global_attrs"]
+        )
+
+        # Save it out to a file
+        with open(output_json_path, "w") as f:
+            f.write(json.dumps(metadata_dict, indent=4))
```

Comment:
Thank you for adding this tab. It was helpful to review!

Reply:
I agree, it was overwhelming before as a single line!